### PR TITLE
Add adr-004 for EntraID

### PR DESCRIPTION
**Pull request by jemnery**

### CODE REVIEW

File: source/documentation/adrs/004-use-entraid-for-identity-and-access-management.html.md.erb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+We will use the existing in-house [EntraID](https://www.microsoft.com/en-us/security/business/identity-access/microsoft-entra-id) (formally Azure Active Directory) service for [Identity and Access Management](https://en.wikipedia.org/wiki/Identity_management) (IDAM), both for DataHub and Find MOJ Data.
+
+Many of our users are already using a `@justice.gov.uk` account as their primary login.
+
+Our users can take advantage of their existing identity to gain access to the Data Catalogue.
+
+## Consequences
+
+- We will no longer need to use IP range restrictions for the services
```
Comment (MatMoore) [inline source/documentation/adrs/004-use-entraid-for-identity-and-access-management.html.md.erb:33]: I think somebody mentioned at standup yesterday that IP restriction will still be kept for the /metrics endpoint, is that right?

File: source/documentation/adrs/004-use-entraid-for-identity-and-access-management.html.md.erb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+- We will not have to run an identity service and managing logging and security of that system
+- We will reduce or eliminate our support requirements for:
+  - joiners, movers and leavers (JML)
+  - issues with multi factor authentication and password resets
+  - name changes (e.g. marriage)
+- Guest accounts are possible, but not managed which means we will need an alternative solution
```
Comment (MatMoore) [inline source/documentation/adrs/004-use-entraid-for-identity-and-access-management.html.md.erb:39]: By guest accounts, do you mean inviting somebody from another organisation to view the catalogue without a justice.gov.uk email? Is this something we have a need for?

**Review by MatMoore — APPROVED** [review]

File: source/documentation/adrs/004-use-entraid-for-identity-and-access-management.html.md.erb (hunk 1 above)
Comment (jemnery) [reply source/documentation/adrs/004-use-entraid-for-identity-and-access-management.html.md.erb:33]: You're right, let me tweak that

File: source/documentation/adrs/004-use-entraid-for-identity-and-access-management.html.md.erb (hunk 2 above)
Comment (jemnery) [reply source/documentation/adrs/004-use-entraid-for-identity-and-access-management.html.md.erb:39]: Good shout. This was heavily based on Rich's EntraID ADR for the AP. It made sense to leave it in at first, but you're right, there's no need to add scope creep here.

**Review by MatMoore — APPROVED** [review]

**Pull request merged**